### PR TITLE
Hubs/Scopes Merge 38 - Use `ScopeType.COMBINED` for cross platform (`InternalSentrySdk`)

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :scroll: Description

To send values from all scope types (global, isolation, current) to cross platform SDKs we use `ScopeType.COMBINED` for `.configureScope` to retrieve all tags, breadcrumbs etc.

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java [hunk 1]
```diff
@@ -44,9 +45,9 @@ public final class InternalSentrySdk {
   @Nullable
   public static IScope getCurrentScope() {
     final @NotNull AtomicReference<IScope> scopeRef = new AtomicReference<>();
-    // TODO [HSM] should this retrieve combined scope?
```
Comment (adinauer) [inline sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java:47]: Yes, at least for cross platform SDKs until they support different scope types separately themselves.

File: sentry/src/main/java/io/sentry/CombinedScopeView.java [hunk 2]
```diff
@@ -431,8 +433,7 @@ public void setPropagationContext(@NotNull PropagationContext propagationContext
 
   @Override
   public @NotNull IScope clone() {
-    // TODO [HSM] just return a new CombinedScopeView with forked scope?
-    return getDefaultWriteScope().clone();
+    return new CombinedScopeView(globalScope, isolationScope.clone(), scope.clone());
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/CombinedScopeView.java:436]: The idea here is to have a clone that doesn't change the original so we're safer by cloning both isolation and current scope.

File: sentry/src/main/java/io/sentry/CombinedScopeView.java [hunk 3]
```diff
@@ -454,7 +455,6 @@ public void bindClient(@NotNull ISentryClient client) {
 
   @Override
   public @NotNull ISentryClient getClient() {
-    // TODO [HSM] checking for noop here doesn't allow disabling via client, is that ok?
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/CombinedScopeView.java:457]: Discussed internally with backend team, if a customer needs this, they can provide their own noop `ISentryClient` that isn't actually `NoOpSentryClient` and thus not detected here.

File: sentry/src/main/java/io/sentry/HubAdapter.java [hunk 4]
```diff
@@ -215,7 +215,6 @@ public void flush(long timeoutMillis) {
 
   @Override
   public @NotNull ISentryLifecycleToken makeCurrent() {
-    // TODO [HSM] this wouldn't do anything since it replaced the current with the same Scopes
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/HubAdapter.java:218]: Yes, at least for cross platform SDKs until they support different scope types separately themselves.

File: sentry/src/main/java/io/sentry/ScopeType.java [hunk 5]
```diff
@@ -4,7 +4,5 @@ public enum ScopeType {
   CURRENT,
   ISOLATION,
   GLOBAL,
-
-  // TODO [HSM] do we need a combined as well so configureScope
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/ScopeType.java:8]: Yes, at least for cross platform SDKs until they support different scope types separately themselves.

File: sentry/src/main/java/io/sentry/Scopes.java [hunk 6]
```diff
@@ -646,9 +646,7 @@ public void withScope(final @NotNull ScopeCallback callback) {
 
     } else {
       final @NotNull IScopes forkedScopes = forkedCurrentScope("withScope");
-      // TODO [HSM] should forkedScopes be made current inside callback?
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/Scopes.java:649]: Yes, I'd expect Sentry.setTag to operate on the forked scopes inside `withScope`

File: sentry-android-core/src/test/java/io/sentry/android/core/InternalSentrySdkTest.kt [hunk 7]
```diff
@@ -138,14 +139,17 @@ class InternalSentrySdkTest {
         )
         // TODO [HSM] add breadcrumbs to all scopes and assert they are there
```
Comment (lbloder) [inline sentry-android-core/src/test/java/io/sentry/android/core/InternalSentrySdkTest.kt:140]: Is this comment still relevant?

Comment (adinauer) [reply sentry-android-core/src/test/java/io/sentry/android/core/InternalSentrySdkTest.kt:140]: No, already removed.

**Review by lbloder — APPROVED** [review]

**Review by markushi — APPROVED** [review]

**Pull request merged**